tools/lint-docstring

```diff
@@ -33,9 +34,47 @@ DOCSTRINGS_FOR_NO_FILE_FOLDERS = {
 BLANK_LINE = f"| {COLUMN_WIDTHS[0]*SPACEBAR}| {COLUMN_WIDTHS[1]*SPACEBAR}| {COLUMN_WIDTHS[2]*SPACEBAR}|\n"
 
 
-def main() -> None:
-    create_file_overview_doc()
-    print(f"Generated {Path(DEVELOPER_DOC).name} successfully.")
+def main(fix_file: bool) -> None:
+    if fix_file:
+        create_file_overview_doc()
+        print(f"Generated {Path(DEVELOPER_DOC).name} successfully.")
+    else:
+        lint_file_overview()
+        print("Successful")
+
+
+def lint_file_overview() -> None:
+    """
+    Used for linting of the developer-file-overview.md document
+    """
+    folder_file_docstring = generate_folder_file_docstrings_dict()
+    computed_doc_dict = extract_docstrings_from_file_overview()
+    missing_entity = ""
+    try:
+        for folder in folder_file_docstring:
+            for file in folder_file_docstring[folder]:
+                missing_entity = file
+                assert (
+                    computed_doc_dict[folder][file]
+                    == folder_file_docstring[folder][file]
+                )
+        # zulipterminal/themes and zulipterminal/scripts are linted using DOCSTRINGS_FOR_NO_FILE_FOLDERS
+        for folder_name in DOCSTRINGS_FOR_NO_FILE_FOLDERS:
+            missing_entity = folder_name
+            assert (
+                computed_doc_dict[folder_name]["no_file_present"]
+                == DOCSTRINGS_FOR_NO_FILE_FOLDERS[folder_name]
+            )
+    except AssertionError:
+        # in case the docstrings do not match
+        print(f"{missing_entity}: docstrings have been changed")
+        print("Run './tools/lint-docstring --fix' to fix")
```

Comment:
Suppose someone changes the markdown file and then runs fix seeing this error, they loose everything they wrote right?

I believe the markdown file is not to be edited after this PR. Should we add a `Generated by code. DO NOT EDIT` warning at the top of the file?

Reply:
@mounilKshah We didn't address this before I did the partial merge, but an extra commit updating the overview file and adjusting the fixing code to take that into account should be doable?